### ISSUE TITLE
Keystore missing in Alfresco Community version 6.2.0-ga

### ISSUE THREAD

**Issue by andreaskring** [Alfresco/acs-deployment#489]
The Java keystore is missing in Alfresco Community version 6.2.0-ga when starting the Alfresco stack with the `docker-compose/community-docker-compose.yml` (after changing the Alfresco version to 6.2.0-ga and the Share version to 6.2.0).

**Steps to reproduce the error**

1. Change the version numbers for Alfresco and Share in `docker-compose/community-docker-compose.yml` like this:
    ```
    services:
        alfresco:
            image: alfresco/alfresco-content-repository-community:6.2.0-ga
            mem_limit: 1500m
    ...
        share:
            image: alfresco/alfresco-share:6.2.0
            mem_limit: 1g
    ...
    ```
    
2. Start the stack (from the `docker-compose` folder) with
    ```
    $ docker-compose -f community-docker-compose.yml up -d
    ```

**Observations**
Tailing the Alfresco log gives this:
```
$ docker-compose -f community-docker-compose.yml logs -f alfresco

alfresco_1            | 2021-01-22 06:59:53,079  ERROR [web.context.ContextLoader] [localhost-startStop-1] Context initialization failed
alfresco_1            | org.alfresco.error.AlfrescoRuntimeException: 00220011 Keystores are invalid
alfresco_1            | 	at org.alfresco.encryption.EncryptionChecker$1.execute(EncryptionChecker.java:78)
alfresco_1            | 	at org.alfresco.encryption.EncryptionChecker$1.execute(EncryptionChecker.java:1)
alfresco_1            | 	at org.alfresco.repo.transaction.RetryingTransactionHelper.doInTransaction(RetryingTransactionHelper.java:450)
alfresco_1            | 	at org.alfresco.repo.transaction.RetryingTransactionHelper.doInTransaction(RetryingTransactionHelper.java:319)
alfresco_1            | 	at org.alfresco.encryption.EncryptionChecker.onBootstrap(EncryptionChecker.java:67)
alfresco_1            | 	at org.springframework.extensions.surf.util.AbstractLifecycleBean.onApplicationEvent(AbstractLifecycleBean.java:56)
alfresco_1            | 	at org.alfresco.repo.management.SafeApplicationEventMulticaster.multicastEventInternal(SafeApplicationEventMulticaster.java:221)
alfresco_1            | 	at org.alfresco.repo.management.SafeApplicationEventMulticaster.multicastEvent(SafeApplicationEventMulticaster.java:186)
alfresco_1            | 	at org.alfresco.repo.management.SafeApplicationEventMulticaster.multicastEvent(SafeApplicationEventMulticaster.java:206)
alfresco_1            | 	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:402)
alfresco_1            | 	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:359)
alfresco_1            | 	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:896)
alfresco_1            | 	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:552)
alfresco_1            | 	at org.springframework.web.context.ContextLoader.configureAndRefreshWebApplicationContext(ContextLoader.java:400)
alfresco_1            | 	at org.springframework.web.context.ContextLoader.initWebApplicationContext(ContextLoader.java:291)
alfresco_1            | 	at org.springframework.web.context.ContextLoaderListener.contextInitialized(ContextLoaderListener.java:103)
alfresco_1            | 	at org.alfresco.web.app.ContextLoaderListener.contextInitialized(ContextLoaderListener.java:70)
alfresco_1            | 	at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4699)
alfresco_1            | 	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5165)
alfresco_1            | 	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
alfresco_1            | 	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:743)
alfresco_1            | 	at org.apache.catalina.core.ContainerBase.access$000(ContainerBase.java:129)
alfresco_1            | 	at org.apache.catalina.core.ContainerBase$PrivilegedAddChild.run(ContainerBase.java:150)
alfresco_1            | 	at org.apache.catalina.core.ContainerBase$PrivilegedAddChild.run(ContainerBase.java:140)
alfresco_1            | 	at java.base/java.security.AccessController.doPrivileged(Native Method)
alfresco_1            | 	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:717)
alfresco_1            | 	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:714)
alfresco_1            | 	at org.apache.catalina.startup.HostConfig.deployDirectory(HostConfig.java:1125)
alfresco_1            | 	at org.apache.catalina.startup.HostConfig$DeployDirectory.run(HostConfig.java:1859)
alfresco_1            | 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
alfresco_1            | 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
alfresco_1            | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
alfresco_1            | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
alfresco_1            | 	at java.base/java.lang.Thread.run(Thread.java:834)
alfresco_1            | Caused by: org.alfresco.error.AlfrescoRuntimeException: 00220010 Failed to create key: metadata
alfresco_1            |  in key store: 
alfresco_1            |    Location: /usr/local/tomcat/shared/classes/alfresco/extension/keystore/keystore
alfresco_1            |    Provider: null
alfresco_1            |    Type:     JCEKS
alfresco_1            | 	at org.alfresco.encryption.AlfrescoKeyStoreImpl.createKey(AlfrescoKeyStoreImpl.java:661)
alfresco_1            | 	at org.alfresco.encryption.AlfrescoKeyStoreImpl.validateKeys(AlfrescoKeyStoreImpl.java:907)
alfresco_1            | 	at org.alfresco.encryption.AlfrescoKeyStoreImpl.validateKeys(AlfrescoKeyStoreImpl.java:188)
alfresco_1            | 	at org.alfresco.encryption.KeyStoreChecker.validateKeyStores(KeyStoreChecker.java:49)
alfresco_1            | 	at org.alfresco.encryption.EncryptionChecker$1.execute(EncryptionChecker.java:73)
alfresco_1            | 	... 33 more
alfresco_1            | Caused by: java.io.FileNotFoundException: /usr/local/tomcat/shared/classes/alfresco/extension/keystore/keystore (No such file or directory)
alfresco_1            | 	at java.base/java.io.FileOutputStream.open0(Native Method)
alfresco_1            | 	at java.base/java.io.FileOutputStream.open(FileOutputStream.java:298)
alfresco_1            | 	at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:237)
alfresco_1            | 	at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:126)
alfresco_1            | 	at org.alfresco.encryption.AlfrescoKeyStoreImpl.getKeyStoreOutStream(AlfrescoKeyStoreImpl.java:396)
alfresco_1            | 	at org.alfresco.encryption.AlfrescoKeyStoreImpl.createKey(AlfrescoKeyStoreImpl.java:645)
alfresco_1            | 	... 37 more
```

I tried to log in to the Alfresco container with
```
$ docker-compose -f community-docker-compose.yml exec -u root alfresco bash
```
and search for "keystore"
```
[root@7e915f538f20 tomcat]# find / -iname '*keystore*'
/usr/share/mime/application/x-java-jce-keystore.xml
/usr/share/mime/application/x-java-keystore.xml
```
but the search did not return any keystore (only the above XML files).

There is no problem if Alfresco Community version `6.2.2-RC1` is used (and Share version `6.2.2-RC3`), but as I understand it from the discussions on the Alfresco Discord channel, there will not be a real (non-RC) release of Community version `6.2.2`, so what is the best way to deal with the above keystore issue for version `6.2.0-ga`?

(I.e. I am a bit reluctant to run a release candidate version (e.g. `6.2.2-RC1`) in production)

**Comment by gavincornwell** [comment]
I would suggest you create a custom image the extends 6.2.0-ga and copies the keystore file to right place within the container. You can find the deafult keystore file in the ZIP distribution.

**Comment by andreaskring** [comment]
@gavincornwell Thank you for your input. IMHO the behavior can be characterized as a bug in a GA release and hence you could consider fixing it and making a hotfix release for the `6.2.0-ga` release? There must be many people having this issue with Alfresco Community version `6.2.0-ga` being the latest GA Community release.

But I will go with you suggestion for now (or generate the keystore with the Java `keytool`).

**Comment by gavincornwell** [comment]
Hi @andreaskring actually it occurred to me over the weekend what the actual issue is...

The docker compose file you're using is targeted at the forthcoming 7.0 release so the configuration is incompatible with 6.2.0, in particular the JAVA_TOOL_OPTIONS environment variable.

If you want to go back to a 6.2.x based system I would suggest comparing your configuration to the one in the old archived repo here: https://github.com/Alfresco/acs-community-deployment/blob/master/docker-compose/docker-compose.yml

I've tried this locally and the system comes up fine with the "6.2.0-ga" image.

Hope that helps!

**Comment by andreaskring** [comment]
Hi @gavincornwell
Thanks for your comment. Initially, I actually experienced the problem using the (acs-deployment) Helm charts for Kubernetes, where I tried to use the `6.2.0-ga` in the [community_values.yml](https://github.com/Alfresco/acs-deployment/blob/master/helm/alfresco-content-services/community_values.yaml#L19), but I guess the reason for this is similar, i.e. that the Helm chart provided in this project is also targeted the forthcoming 7.0 release.

I will try to look into the `JAVA_TOOL_OPTIONS` env variable to see, if this will resolve the issue.
Thanks a lot for your help
